compiler/src/org.graalvm.compiler.replacements.aarch64/src/org/graalvm/compiler/replacements/aarch64/AArch64GraphBuilderPlugins.java

```diff
@@ -109,45 +99,33 @@ public boolean apply(GraphBuilderContext b, ResolvedJavaMethod targetMethod, Rec
         r.registerMethodSubstitution(substituteDeclaringClass, "bitCount", type);
     }
 
-    private static void registerMathPlugins(InvocationPlugins plugins) {
-        Registration r = new Registration(plugins, Math.class);
-        registerUnaryMath(r, "sin", SIN);
-        registerUnaryMath(r, "cos", COS);
-        registerUnaryMath(r, "tan", TAN);
-        registerUnaryMath(r, "exp", EXP);
-        registerUnaryMath(r, "log", LOG);
-        registerUnaryMath(r, "log10", LOG10);
-        r.register2("pow", Double.TYPE, Double.TYPE, new InvocationPlugin() {
-            @Override
-            public boolean apply(GraphBuilderContext b, ResolvedJavaMethod targetMethod, Receiver receiver, ValueNode x, ValueNode y) {
-                b.push(JavaKind.Double, b.append(BinaryMathIntrinsicNode.create(x, y, BinaryMathIntrinsicNode.BinaryOperation.POW)));
-                return true;
-            }
-        });
-        registerRound(r, "rint", RoundingMode.NEAREST);
-        registerRound(r, "ceil", RoundingMode.UP);
-        registerRound(r, "floor", RoundingMode.DOWN);
-    }
-
-    private static void registerUnaryMath(Registration r, String name, UnaryOperation operation) {
-        r.register1(name, Double.TYPE, new InvocationPlugin() {
-            @Override
-            public boolean apply(GraphBuilderContext b, ResolvedJavaMethod targetMethod, Receiver receiver, ValueNode value) {
-                b.push(JavaKind.Double, b.append(UnaryMathIntrinsicNode.create(value, operation)));
-                return true;
-            }
-        });
-    }
-
-    private static void registerRound(Registration r, String name, RoundingMode mode) {
-        r.register1(name, Double.TYPE, new InvocationPlugin() {
-            @Override
-            public boolean apply(GraphBuilderContext b, ResolvedJavaMethod targetMethod, Receiver receiver, ValueNode arg) {
-                b.push(JavaKind.Double, b.append(new AArch64RoundNode(arg, mode)));
-                return true;
-            }
-        });
-    }
+    /*
```

Comment:
So, looking at implementing `AArch64MathPowOp` by emitting `libm#pow` and I immediately run into an issue:

```
000000000000c2c0 <pow10l@GLIBC_2.17>:
    c2c0:       9e660000        fmov    x0, d0
    c2c4:       9eae0001        fmov    x1, v0.d[1]
```

Our assembler does not seem to have a way to represent `v0.d[1]` which should take the top 64-bits from the 128-bit `v0` SIMD register.

I could certainly figure out how to augment the assembler, but it I suspect I simply don't know what I'm doing. If @theRealAph or @adinn or someone can confirm that the `AArch64Assembler` lacks this specific variant of of `fmov`, I'd be glad to truck along and give it a go.

Additionally, is it even correct to assume we need to transcode the `libm.so` implementation? It seems to be what the `AMD64` variant of `MathPowOp` is doing.

Reply:
Likewise, `log10`:

```
    cc30:       6f00e401        movi    v1.2d, #0x0
```

I dunno if we support the `v1.2d` style register.  I think that's all 128 bits of the `v1` register? Any reason we wouldn't just call it `v1' though?

Reply:
Yes, you should call it v1. But I don't quite understand what you are trying to do here. It makes sense for the time being to call the routine in libm, because we are at present working on a custom AArch64 implementation of Math.pow() in HotSpot. When that's done you can import it.